.github/workflows/publish-release.yaml

```diff
@@ -0,0 +1,44 @@
+name: Gingersnap db-syncer - Create tag and Publish
+
+on:
+  workflow_call:
+    inputs:
+      ref:
+        type: string
+        default: main
+      repo: 
+        type: string
+        default: gingersnap-project/db-syncer
+        
+      
+env:
+  GITHUB_TOKEN: ${{ secrets.PAT_TOKEN }}  
+
+jobs:
+  tag:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Configure Git
+        run: |
+          git config --global user.name "infinispan-qe-bot"
+          git config --global user.email "q*@infinispan.org"
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with: 
+          ref: ${{ inputs.ref }} 
+          repository: ${{ inputs.repo }}
+          
+      - name: Tag the release
+        run: |
+         git tag -a v${{ inputs.ref }} -m "Release db-syncer:: ${{ inputs.ref }}"
```

Comment:
We're using `intputs.ref` for too many things here. Currently unless a branch already exists with name `inputs.ref`, then the checkout step will fail.

I think we need to add a third input `tag`. `inputs.tag` is used to create the tag, whereas `inputs.ref` is used for the `ref` field in the checkout action.